collect_app/src/main/java/org/odk/collect/android/widgets/BearingWidget.java

```diff
@@ -73,34 +77,49 @@ public void onClick(View v) {
                         .getActivityLogger()
                         .logInstanceAction(this, "recordBearing", "click",
                                 getFormEntryPrompt().getIndex());
-                Intent i;
-                i = new Intent(getContext(), BearingActivity.class);
 
-                waitForData();
-                ((Activity) getContext()).startActivityForResult(i,
-                        RequestCodes.BEARING_CAPTURE);
+                if (isSensorAvailable) {
+                    Intent i;
+                    i = new Intent(getContext(), BearingActivity.class);
```

Comment:
You can do that in one line:
`Intent i = new Intent(getContext(), BearingActivity.class);`